.github/workflows/test-inference.yml

```
@@ -0,0 +1,34 @@
+name: Test inference
+
+on: 
+  pull_request:    
+  push:
+    branches:
+      - main
+
+jobs:
+  test:  
+    name: "Test inference"  
+    # This action is designed only to run on the Stability research cluster at this time, so many assumptions are made about the envrionment
+    if: github.repository == 'stability-ai/generative-models'
+    runs-on: [self-hosted, slurm, g40]
+    steps:
+      - uses: actions/checkout@v3
+      - name: "Symlink checkpoints"
+        run: ln -s /admin/home-palp/sgm-checkpoints checkpoints
+      - name: "Setup python"
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"

```

Might want to cache the dependencies (though this might be unnecessary if your runner already has a good local cache)
```suggestion
          python-version: "3.10"
          cache: pip
          cache-dependency-path: requirements-pt2.txt
```

---

Yeah, these runners are not that ephemeral so it has a good chance of being on disk. We also have our own pass-through cache I can hook up that will keep it within AWS.

---

Actually, I've changed my mind and am moving this to a shared path now, which we'll point to via a secret - also making the action a bit closer to being reusable elsewhere.